playbooks/roles/demo/tasks/deploy.yml

```diff
@@ -10,7 +10,7 @@
   register: demo_checkout
 
 - name: import demo course
-  shell: "{{ demo_edxapp_venv_bin }}/python ./manage.py cms --settings={{ demo_edxapp_settings }} import {{ demo_edxapp_course_data_dir }} {{ demo_code_dir }}"
+  shell: ". {{ edxapp_app_dir }}/edxapp_env && {{ demo_edxapp_venv_bin }}/python ./manage.py cms --settings={{ demo_edxapp_settings }} import {{ demo_edxapp_course_data_dir }} {{ demo_code_dir }}"
```

Comment:
I really don't like this change. By reusing the `edxapp_app_dir` variable, we're making the `demo` role dependent on the `edxapp` role. From now on, if the `demo` role is run in a playbook without the `edxapp` role, this task is going to fail. 

I think the proper "Ansible" way of doing this is to construct a new variable in `playbooks/roles/demo/defaults/main.yml` like:

```
demo_edxapp_env: '{{ COMMON_APP_DIR }}/{{ demo_edxapp_user }}/edxapp_env'
```

then rewrite the task as:

```
- name: import demo course
  shell: ".{{ demo_edxapp_env }} && {{ demo_edxapp_venv_bin }}/python ...
```
This way, the default `demo_` variables in the `demo` role match up with the default `edxapp_` variables in the `edxapp` role. It's still not ideal, but at least the dependency has been removed.

@thraxil do you think I'm being too pedantic here?

Reply:
Thanks @tkeemon. I've already contributed this upstream (https://github.com/edx/configuration/pull/4812) so it's a bit late to do changes here. But thanks!

Reply:
@tkeemon I'm open to fixing this upstream, do you think it's needed?